repo/splitter/splitter_rabinkarp64.go

```diff
@@ -23,23 +23,69 @@ func (rs *rabinKarp64Splitter) Reset() {
 	rs.count = 0
 }
 
-func (rs *rabinKarp64Splitter) ShouldSplit(b byte) bool {
-	rs.rh.Roll(b)
-	rs.count++
+func (rs *rabinKarp64Splitter) NextSplitPoint(b []byte) int {
+	var fastPathBytes int
 
-	if rs.rh.Sum64()&rs.mask == 0 && rs.count >= rs.minSize {
-		rs.count = 0
-		return true
+	// simple optimization, if we're below minSize, there's no reason to even
+	// look at the Sum64() or maxSize
+	if left := rs.minSize - rs.count - 1; left > 0 {
+		fastPathBytes = left
+		if fastPathBytes > len(b) {
+			fastPathBytes = len(b)
+		}
+
```

Comment:
`buzhash32Splitter` has an unrolled loop here. Can it be done here as well or is there a restriction for RK64, or it does not provide any benefits?

Reply:
done